pkg/providers/v1/tags.go

```diff
@@ -154,6 +154,31 @@ func (t *awsTagging) hasClusterTag(tags []ec2types.Tag) bool {
 	return false
 }
 
+// hasClusterTagOwned checks if the resource has the cluster tag with the value "owned"
+// This is used to determine if the resource is owned by the controller.
+// It checks both legacy and new tags.
+func (t *awsTagging) hasClusterTagOwned(tags []ec2types.Tag) bool {
+	if len(t.ClusterID) == 0 {
+		return false
```

Comment:
i asked this on the #1158 review, but just curious if this allows the possibility for hiding an error?